packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -576,24 +690,49 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
     }
 
     private getTimeForX(x: number): bigint {
```

Comment:
It seems getTimeForX and getXForTime could be simplified if we always use this.getChartWidth(). You wouldn't even need to do a null check on the chart then.

When I debug, this.getChartWidth() and this.chartRef.current.chartInstance.width are almost always the same.

Sometimes they are not for a short time because the latter gets delayed update after a draw, I'm not even sure which width we should be using. And sometimes the chartInstance is undefined for a short time, perhaps during chart.js internal update. Also if you quickly resize the sash so that there is no room for the chart, this.getChartWidth() returns 0 but  this.chartRef.current.chartInstance.width gets stuck at the last visible non-zero width.